packages/aws-cdk-lib/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -180,6 +181,14 @@ export interface ApplicationLoadBalancedServiceBaseProps {
    */
   readonly loadBalancer?: IApplicationLoadBalancer;
 
+  /**
+   * The application load balancer's listener that traffic should be served over.
+   * Only has an effect when specifying `loadBalancer`.
```

Comment:
It would be great to understand the rationale for this condition.

Also, since this takes effect only for a specific value, let's add some input validation in the constructor to ensure that the provided listener specifies `loadBalancer`. This will prevent the scenario where users provide a listener which silently has no effect, and they have no visibility into why their setup does not work.

Reply:
Primarily because if you're specifying a custom listener you would also need to provide a custom ALB for it to work, as far as I could tell. If you try to put a custom listener with a non-custom load balancer, it should fail because the listener comes from a different ALB. If that's wrong I'm happy to walk that bit back, but I'm also happy to go about putting in the validation to ensure that's all correctly set

Reply:
Would this satisfy your request for an input validation rule?
```ts
    if (props.listener && !props.loadBalancer) {
      throw new ValidationError('If you specify listener, you must also specify loadBalancer.', this);
    }
```

Reply:
> Primarily because if you're specifying a custom listener you would also need to provide a custom ALB for it to work

If the listener is attached to the ALB anyway, would it be better to just use `loadBalancer.listeners` to get the attached listener for the custom ALB provided?

This way there is no risk of the user providing a listener which is not attached to the provided load balancer.

Reply:
> > Primarily because if you're specifying a custom listener you would also need to provide a custom ALB for it to work
> 
> If the listener is attached to the ALB anyway, would it be better to just use `loadBalancer.listeners` to get the attached listener for the custom ALB provided?
> 
> This way there is no risk of the user providing a listener which is not attached to the provided load balancer.

I guess my concern there is what if you're providing a load balancer that has one listener it actually needs to use and one that's used for a different service or an HTTP->HTTPS redirect or similar.

Reply:
That's fair but also doesn't address the issue of making sure it uses a SPECIFIC listener as opposed to, say, the first one it finds. That's my requirement on this, I want to be able to give it a specific listener on an LB that might have two or three

Reply:
Has my reply there helped explain what I'm trying to solve for here? If not I can provide some level of a basic example of the code I've got to back it up.